src/lib/y2network/config_reader/wicked.rb

```diff
@@ -0,0 +1,41 @@
+# Copyright (c) [2019] SUSE LLC
+#
+# All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of version 2 of the GNU General Public License as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+require "y2network/config"
+
+Yast.import "LanItems"
+
+module Y2Network
+  module ConfigReader
+    # This class reads the current configuration from the system
+    class Wicked
+      # @return [Y2Network::Config] Network configuration
+      def config
+        Yast::LanItems.Read
+        Config.new(interfaces: interfaces)
+      end
+
+      def interfaces
+        # TODO: for the time being, we are just relying in the underlying stuff.
+        Yast::LanItems.current_device_names.map do |name|
```

Comment:
I think that ```Wicked``` reader should be build on top of ```NetworkInterfaces``` bcs ```LanItems``` is higher level. At least it collects configured even unconfigured devices together with udev rules and some more things. This can be replaced e.g. by ```NetworkInterfaces::List``` or ```NetworkInterfaces::Filter``` depending of what you need.